### PR TITLE
Feature/lazy loading modification

### PR DESCRIPTION
**Pull request by DarinHajou**
Follow up on the implementation for the lazy loading modifications in the backend. Waiting for Joel to review and outline the next steps.

### CODE REVIEW

**Comment by swoocn** [comment]
Hi @DarinHajou - keep in mind that the `fetchSellers` API call is also invoked after a user submits a search query.  Therefore, we also need to pass the bounds as an argument, otherwise, an empty bounds.  The build is currently failing because of this fact ->

image

Can you please resolve accordingly?  Thx!

File: src/services/sellerApi.ts
Comment (swoocn) [inline src/services/sellerApi.ts:6]: This is going to impact how we're executing the search.

**Comment by DarinHajou** [comment]
> Hi @DarinHajou - keep in mind that the `fetchSellers` API call is also invoked after a user submits a search query. Therefore, we also need to pass the bounds as an argument, otherwise, an empty bounds. The build is currently failing because of this fact ->
> 
> image
> 
> Can you please resolve accordingly? Thx!


Yup, I know. I was waiting for a reply from Joel. I'll get it done tomorrow. Thanks!

**Comment by DarinHajou** [comment]
Ready to merge after review!

**Comment by DarinHajou** [comment]
Ready to merge after review!

File: src/components/shared/SearchBar/SearchBar.tsx
Comment (swoocn) [inline src/components/shared/SearchBar/SearchBar.tsx]: Hi @DarinHajou - have you had a chance to test this section?  It seems that the `mapInstance` is consistently null, which is causing the bounds error **[screenshot attached below]**.  As a result, the submit search functionality isn't quite working as it did before.  When you have a moment, could you and/ or @joey364 please take a look at this? 
 Thanks!
image

**Comment by DarinHajou** [comment]
> Hi @DarinHajou - keep in mind that the `fetchSellers` API call is also invoked after a user submits a search query. Therefore, we also need to pass the bounds as an argument, otherwise, an empty bounds. The build is currently failing because of this fact ->
> 
> image
> 
> Can you please resolve accordingly? Thx!

I'll get this done, I'll let you know soon as possible.

File: src/components/shared/SearchBar/SearchBar.tsx
Comment (swoocn) [reply src/components/shared/SearchBar/SearchBar.tsx]: Hi again @DarinHajou - 
After further testing, it seems we are still triggering the fetching of sellers with even slight map movements, which is placing quite a load on the fetchSellers API endpoint.  I’m not sure how this improves performance compared to the origin/radius approach we had earlier, where we could control additional fetching.  Do we have the same level of control with the bounding box approach?  I might be missing something here—please let me know if I am.

Also, it looks like we're fetching a maximum of 36 sellers, even without performing a search.  It seems 36 is the limit we can display on the map, rather than pulling all sellers within the bounds.  You may want to confirm with our PM, but I believe the 36-limit + [review count + recent reviews] applies to search results only.  We should still display all  sellers even when no search is submitted.

Comment (swoocn) [reply src/components/shared/SearchBar/SearchBar.tsx]: > Also, it looks like we're fetching a maximum of 36 sellers, even without performing a search. It seems 36 is the limit we can display on the map, rather than pulling all sellers within the bounds. You may want to confirm with our PM, but I believe the 36-limit + [review count + recent reviews] applies to search results only. We should still display all sellers even when no search is submitted.

Please disregard this comment @DarinHajou.  Our PM confirmed to maintain the limit, even with no searches.  Thx!

**Review by swoocn — APPROVED** [review]
LGTM @DarinHajou; another great effort! 😎👍
Made some minor adjustments but everything seemed to test out fine in the sandbox environment.  We'll see how it goes as we propagate the deployment.
Thanks again for your hard work!

**Pull request merged**
